lib/kube/proxy/server.go

```diff
@@ -103,23 +108,43 @@ type TLSServer struct {
 	heartbeats   map[string]*srv.Heartbeat
 	closeContext context.Context
 	closeFunc    context.CancelFunc
+	// watcher monitors changes to kube cluster resources.
+	watcher *services.KubeClusterWatcher
+	// reconciler reconciles proxied kube clusters with kube_clusters resources.
+	reconciler *services.Reconciler
+	// monitoredKubeClusters contains all kube clusters the proxied kube_clusters are
+	// reconciled against.
+	monitoredKubeClusters monitoredKubeClusters
+	// reconcileCh triggers reconciliation of proxied kube_clusters.
+	reconcileCh chan struct{}
+	log         *logrus.Entry
 }
 
 // NewTLSServer returns new unstarted TLS server
 func NewTLSServer(cfg TLSServerConfig) (*TLSServer, error) {
 	if err := cfg.CheckAndSetDefaults(); err != nil {
 		return nil, trace.Wrap(err)
 	}
+	log := cfg.Log.WithFields(logrus.Fields{
+		trace.Component: cfg.Component,
+	})
 	// limiter limits requests by frequency and amount of simultaneous
 	// connections per client
 	limiter, err := limiter.NewLimiter(cfg.LimiterConfig)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	fwd, err := NewForwarder(cfg.ForwarderConfig)
-	if err != nil {
+
+	fwd, err := NewForwarder(log, cfg.ForwarderConfig)
+	// if load kubeconfig returned an error but the service is willing to start
+	// with resource watcher we let the service continue
+	if err != nil && (!trace.IsBadParameter(err) ||
+		cfg.KubeServiceType != KubeService ||
```

Comment:
Does the NewForwarder returns a valid fwd in case of `err != nil`  ? 
Can we rework this inside NewForwarder instead of introducing unnatural flow ?